ggml-metal.metal

```diff
@@ -428,6 +442,32 @@ inline float block_q_n_dot_y(device const block_q4_1 * qb_curr, float sumy, thre
     return d * (acc[0] + acc[1]) + sumy * m;
 }
 
+// function for calculate inner product between half a q5_0 block and 16 floats (yl), sumy is SUM(yl[i])
+// il indicates where the q5 quants begin (0 or QK5_0/4)
+// we assume that the yl's have been multiplied with the appropriate scale factor
+// that corresponds to the missing bit shifts (1, 1/16, 1/256, 1/4096)
+inline float block_q_n_dot_y(device const block_q5_0 * qb_curr, float sumy, thread float * yl, int il) {
+    float d = qb_curr->d;
+    float2 acc = 0.f;
+    device const uint16_t * qs = ((device const uint16_t *)qb_curr + 3 + il/2);
```

Comment:
```suggestion
    device const uint16_t * qs = ((device const uint16_t *)qb_curr + 1 + il/2);
```

Reply:
+3 should be correct because it skip 4 uint8_t (qh). Am I wrong?

Reply:
Yes, sorry I missed that